src/main/java/nextstep/subway/line/domain/Lines.java

```diff
@@ -0,0 +1,56 @@
+package nextstep.subway.line.domain;
+
+import nextstep.subway.HibernateUtils;
+import nextstep.subway.path.dto.PathResponse;
+import nextstep.subway.station.domain.Station;
+import nextstep.subway.station.dto.StationResponse;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.WeightedMultigraph;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class Lines {
+
+    private List<Line> lines;
+
+    public Lines(List<Line> lines) {
+        this.lines = lines;
+    }
+
+    public PathResponse findPath(Station source, Station target) {
+        DijkstraShortestPath dijkstraShortestPath = new DijkstraShortestPath<>(toGrapth());
+        List<Station> shortestPath = dijkstraShortestPath.getPath(source, target).getVertexList();
+        int distance = IntStream.range(0, shortestPath.size())
+                .reduce((accum, idx) -> {
+                    Station upStation = shortestPath.get(idx - 1);
+                    Station downStation = shortestPath.get(idx);
+                    Section section = this.findSection(upStation, downStation);
+                    return accum + section.getDistance();
+                })
+                .getAsInt();
+        return new PathResponse(distance, shortestPath.stream().map(StationResponse::new).collect(Collectors.toList()));
+    }
+
+    private Section findSection(Station upStation, Station downStation) {
+        return this.lines.stream().filter(line -> line.getSections().findSection(upStation, downStation).isPresent())
```

Comment:
getter를 사용하지 않고 로직을 구현해보아요
그리고 한줄에 다 쓰지 말고 한줄에 한 기능씩 작성해보아요

```suggestion
        return this.lines.stream()
            .filter(line -> line.existsSection(upStation, downStation)) // 비즈니스 용어에 적합한 메서드명으로 바꿔보는게 좋겠네요
```